optuna/batch/study.py

```diff
@@ -0,0 +1,197 @@
+import copy
+from functools import partial
+import gc
+import math
+import types
+from typing import Any
+from typing import Callable
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import Type
+from typing import Union
+
+import numpy as np
+
+import optuna
+from optuna._experimental import experimental
+import optuna.logging
+
+ObjectiveFuncType = Callable[["optuna.batch.trial.BatchTrial"], np.ndarray]
+CallbackFuncType = Callable[["optuna.study.Study", "optuna.trial.FrozenTrial"], None]
+
+_logger = optuna.logging.get_logger(__name__)
+
+
+class BatchStudy(object):
+    def __init__(self, study: "optuna.study.Study", batch_size: int):
+        self._study = study
+        self._batch_size = batch_size
+
+    def __getattr__(self, attr_name: str) -> Any:
+        return getattr(self._study, attr_name)
+
+    @property
+    def batch_size(self) -> int:
+        """Return the size of batches.
+
+        Returns:
+            Size of batches.
+        """
+
+        return self._batch_size
+
+    def optimize(
+        self,
+        objective: ObjectiveFuncType,
+        timeout: Optional[int] = None,
+        n_batches: Optional[int] = None,
```

Comment:
Just an idea but existing users of the framework might be more familiar with `n_trials` (and the number of batches can be derived from the total number of trials). It'd also make the interface similar the existing one and thus less to memorize.

Reply:
That makes sense. I replaced it with `n_trials` in https://github.com/optuna/optuna/pull/1729/commits/e4cdf6c7489a56c1c165ff54399163231adb6400.